CI_README.md

```diff
@@ -0,0 +1,51 @@
+If you're interested in developing the repositories that Magic Modules manages, here are the things you'll want to know.
+
+## What The Magician Does
+The Magician takes the PR you write against Magic Modules and creates the downstream (generated) commits that are a result of your changes.  It posts the diffs created by your changes as a comment on your PR, to aid in review.  When your PR is merged, it updates the downstream repositories with your changes.
+
+## Your Workflow
+
+You'll write some code and open a PR.  The Magician will run the generator and the downstream tests.  If either the generator or the tests fail, you'll see "check failed" status on your PR, which will link to the step that failed.  Once all the generation steps succeed, the Magician will comment on your PR with links to review the generated diffs.  Your reviewer will review those diffs (as well as your code).  Once your PR is approved, simply merge it in - the Magician will ensure that the downstream repositories are updated within about 20 minutes.
+
+# CI For CI Developers
+If you're working on enhancing these CI tools, you'll need to know about the structure of the pipeline and about how to develop and test.
+
+## How the pipeline works
+The pipeline is written in Github Actions, and is defined in the workflow .yml files in the .github/workflows directory of this repository.  The documentation on those files is located [here](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/building-actions).  Documentation on the substeps is located below.
+
+### Generation & Diffing
+The generation / diff pipeline has one Action per downstream.  It generates the downstream at the PR's merge commit and at the left-side parent (`HEAD~`), which guarantees isolation of exclusively the changes made by the PR in question.  It creates commits for the downstreams as before-and-after commits.  Those commits don't have relevant git history - they're not meant to be applied - but they are used in a [two-dot diff](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-comparing-branches-in-pull-requests#three-dot-and-two-dot-git-diff-comparisons).  This means that if there are no changes to the downstream, you'll see an empty diff.  These downstream branches are named `pr-$number-old` and `pr-$number-new` respectively.
+
+### Downstream Pushing
+The Magician maintains a tag called `downstream-master` that tracks the Magic Modules commit the downstreams are up to date with.
+
+Every 20 minutes, the downstreams are synced with the MM repo. Commit-by-commit for each commit since `downstream-master`, the Magician runs MM on a local copy of each downstream. Any changes MM generates are committed, and that commit's message will link back to the MM commit it came from.
+
+In effect, this means that each downstream commit will correspond 1:1 to an MM commit. If an MM commit had no changes in a downstream, no commit will be created.
```

Comment:
No more merging all MM commits into a single downstream? Does this mean we should be enforcing squash commits from branches into MM?

Reply:
I think we should have squash-merge as the only MM merge option by default, and turn on rebase merges when merging long-lived branches like `3.0.0` (where every commit is already a squash merge).

That way, each downstream commit is 1:1 with a PR.